dbmigrator/tests/test_cli.py

```diff
@@ -246,3 +247,73 @@ def test_mark_as_false_already_false(self):
         stdout = out.getvalue()
         self.assertIn('20160228202637_add_table    False', stdout)
         self.assertIn('20160228212456_cool_stuff   False', stdout)
+
+
+class GenerateTestCase(BaseTestCase):
+    @mock.patch('dbmigrator.utils.timestamp')
+    def test(self, timestamp):
+        timestamp.return_value = '20160423231932'
+        filename = '{}_a_new_migration.py'.format(timestamp())
+        expected_path = os.path.join(
+            testing.test_migrations_directories[0], filename)
+
+        def cleanup():
+            if os.path.exists(expected_path):
+                os.remove(expected_path)
+        self.addCleanup(cleanup)
+
+        testing.install_test_packages()
+
+        with testing.captured_output() as (out, err):
+            self.target(['--context', 'package-a', 'generate',
+                         'a_new_migration'])
+
+        stdout = out.getvalue()
+        stderr = err.getvalue()
+
+        self.assertEqual(
+            'Generated migration script "dbmigrator/tests/data/package-a/'
+            'package_a/migrations/{}"\n'.format(filename),
+            stdout)
+        self.assertEqual('', stderr)
+
+        self.assertTrue(os.path.exists(expected_path))
+
+        with open(expected_path, 'r') as f:
+            content = f.read()
+
+        self.assertIn('# -*- coding: utf-8 -*-', content)
+        self.assertIn('def up(cursor):', content)
+        self.assertIn('def down(cursor):', content)
+
+    def test_no_migrations_directory(self):
+        with self.assertRaises(Exception) as cm:
+            self.target(['generate', 'a_new_migration'])
+
+        self.assertEqual('migrations directory undefined',
+                         str(cm.exception))
+
+    def test_multiple_migrations_directory(self):
+        with self.assertRaises(Exception) as cm:
+            self.target(['--migrations-directory', '/tmp',
+                         '--migrations-directory', '/var/tmp',
+                         'generate', 'a_new_migration'])
+
+        self.assertEqual('more than one migrations directory specified',
+                         str(cm.exception))
+
+    @mock.patch('dbmigrator.utils.timestamp')
+    def test_migrations_directory_does_not_exist(self, timestamp):
+        timestamp.return_value = '20160423231932'
+        filename = '{}_a_new_migration.py'.format(timestamp())
+        expected_path = '/tmp/dbmigrator-tests/m/{}'.format(filename)
```

Comment:
Possibly better to use the standard library's `tempfile` for windows compatibility? But then again, I see you ref `/tmp` in many places. So maybe something to address when and if the need arises.